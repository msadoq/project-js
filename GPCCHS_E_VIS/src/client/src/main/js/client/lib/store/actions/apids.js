import simple from 'store/helpers/simpleActionCreator';
import {
  WS_APIDS_ASK,
  WS_APIDS_ADD,
} from '../types';

export const askApids = simple(
  WS_APIDS_ASK,
  'domainId',
  'sessionId'
);

export const addApids = simple(
  WS_APIDS_ADD,
  'tupleId',
  'apids'
);
