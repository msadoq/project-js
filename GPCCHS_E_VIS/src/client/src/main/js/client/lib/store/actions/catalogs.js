import simple from 'store/helpers/simpleActionCreator';
import {
  WS_CATALOGS_ASK,
  WS_CATALOGS_ADD,
  WS_CATALOG_ITEMS_ASK,
  WS_CATALOG_ITEMS_ADD,
  WS_COM_OBJECTS_ASK,
  WS_COM_OBJECTS_ADD,
  WS_UNIT_ASK,
  WS_UNIT_ADD,
  WS_ASK_CATALOGS_AND_ITEMS_AND_UNIT,
} from '../types';

export const askCatalogs = simple(
  WS_CATALOGS_ASK,
  'domainId',
  'sessionId'
);

export const addCatalogs = simple(
  WS_CATALOGS_ADD,
  'tupleId',
  'catalogs'
);

export const askCatalogItems = simple(
  WS_CATALOG_ITEMS_ASK,
  'domainId',
  'sessionId',
  'name'
);

export const addCatalogItems = simple(
  WS_CATALOG_ITEMS_ADD,
  'tupleId',
  'name',
  'items'
);

export const askComObjects = simple(
  WS_COM_OBJECTS_ASK,
  'domainId',
  'sessionId',
  'name',
  'itemName'
);

export const addComObjects = simple(
  WS_COM_OBJECTS_ADD,
  'tupleId',
  'name',
  'itemName',
  'comObjects'
);

export const askUnit = simple(
  WS_UNIT_ASK,
  'domainId',
  'sessionId',
  'name',
  'itemName'
);

export const addUnit = simple(
  WS_UNIT_ADD,
  'tupleId',
  'name',
  'itemName',
  'unit'
);

export const askCatalogsAndItemsAndUnit = simple(
  WS_ASK_CATALOGS_AND_ITEMS_AND_UNIT,
  'domainId',
  'sessionId',
  'name',
  'itemName'
);
