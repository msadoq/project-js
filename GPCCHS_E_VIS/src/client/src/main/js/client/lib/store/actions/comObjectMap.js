import simple from 'store/helpers/simpleActionCreator';
import {
  SET_COM_OBJECT_MAP,
} from '../types';

const setComObjectMap = simple(
  SET_COM_OBJECT_MAP,
  'newComObjectMap'
);

export default setComObjectMap;
