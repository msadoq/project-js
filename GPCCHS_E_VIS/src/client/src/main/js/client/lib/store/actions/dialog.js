import simple from '../helpers/simpleActionCreator';
import * as types from '../types';

const openDialog = simple(types.HSC_OPEN_DIALOG, 'windowId', 'id', 'type', 'options');

const closeDialog = simple(
  types.HSC_CLOSE_DIALOG,
  'windowId',
  'id',
  'choice'
);

export default {
  openDialog,
  closeDialog,
};
