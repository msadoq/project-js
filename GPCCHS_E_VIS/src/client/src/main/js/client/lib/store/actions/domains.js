import simple from '../helpers/simpleActionCreator';
import * as types from '../types';

export default {
  updateDomains: simple(types.HSS_UPDATE_DOMAINS, 'domains'),
};
