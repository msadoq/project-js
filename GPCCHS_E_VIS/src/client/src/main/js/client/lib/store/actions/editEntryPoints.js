/* eslint-disable import/prefer-default-export */
import simple from '../helpers/simpleActionCreator';
import { WS_VIEW_UPDATE_ENTRYPOINT_NAME } from '../types';


export const updateEntryPointName =
  simple(WS_VIEW_UPDATE_ENTRYPOINT_NAME, 'viewId', 'epId', 'epName');
