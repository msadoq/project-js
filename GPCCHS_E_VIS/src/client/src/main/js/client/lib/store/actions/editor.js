import * as types from '../types';
import simple from '../helpers/simpleActionCreator';

export const openHtmlEditor = simple(types.WS_WINDOW_OPEN_HTML_EDITOR, 'viewId');
export const closeHtmlEditor = simple(types.WS_WINDOW_CLOSE_HTML_EDITOR);
