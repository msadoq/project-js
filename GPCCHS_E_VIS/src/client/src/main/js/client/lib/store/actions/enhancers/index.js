import ifPathChanged from './ifPathChanged';
import addUuids from './addUuids';
import withProfiling from './withProfiling';

export default {
  ifPathChanged,
  addUuids,
  withProfiling,
};
