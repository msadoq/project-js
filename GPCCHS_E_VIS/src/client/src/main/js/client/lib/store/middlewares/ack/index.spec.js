import configureMockStore from 'redux-mock-store';
import makeAckMiddleware from './index';
import state from '../../../common/jest/mocks/stateAlarms.json';
import { sendAlarmAck } from '../../../viewManager/GroundAlarmView/store/actions';

const unknownAction = { type: 'UNKNOWN' };

const fakeRequestAck = (tbdId, dataId, alarms, cb) => {
  if (alarms[0].oid === 'ERROR') {
    return cb('ERROR');
  }
  if (alarms[0].oid === 'TIMEOUT') {
    return undefined;
  }
  return cb();
};

const mockStore = configureMockStore([
  makeAckMiddleware(fakeRequestAck),
]);

describe('ack:middleware', () => {
  jest.useFakeTimers();
  test('it does nothing with unknown actions', () => {
    const store = mockStore(state);
    store.dispatch(unknownAction);
    expect(store.getActions()).toEqual([unknownAction]);
  });

  ['gma', 'oba'].forEach((ackType) => {
    const describeText = ackType === 'gma' ? 'GroundMonitoringAlarm' : 'OnBoardAlarm';
    describe(describeText, () => {
      const viewId = ackType === 'gma' ? Object.keys(state.GroundAlarmViewData)[0] : Object.keys(state.OnboardAlarmViewData)[0];
      test('simple ack success', () => {
        const store = mockStore(state);
        const alarms = [{ oid: 1 }];
        store.dispatch(sendAlarmAck(viewId, 'ACKID', alarms, 'COMMENT', ackType));
        expect(store.getActions()).toMatchSnapshot();
      });
      test('several ack success', () => {
        const store = mockStore(state);
        const alarms = [
          { oid: 1 },
          { oid: 2 },
          { oid: 3 },
          { oid: 4 },
        ];
        store.dispatch(sendAlarmAck(viewId, 'ACKID', alarms, 'COMMENT', ackType));
        expect(store.getActions()).toMatchSnapshot();
      });
      test('several ack failure', () => {
        const store = mockStore(state);
        const alarms = [
          { oid: 'ERROR' },
          { oid: 'ERROR' },
          { oid: 'ERROR' },
          { oid: 'ERROR' },
        ];
        store.dispatch(sendAlarmAck(viewId, 'ACKID', alarms, 'COMMENT', ackType));
        expect(store.getActions()).toMatchSnapshot();
      });
      test('several ack timeout', () => {
        const store = mockStore(state);
        const alarms = [
          { oid: 'TIMEOUT' },
          { oid: 'TIMEOUT' },
          { oid: 'TIMEOUT' },
          { oid: 'TIMEOUT' },
        ];
        store.dispatch(sendAlarmAck(viewId, 'ACKID', alarms, 'COMMENT', ackType));
        jest.runAllTimers();
        expect(store.getActions()).toMatchSnapshot();
      });
      test('mixed several success, failure and timeout', () => {
        const store = mockStore(state);
        const alarms = [
          { oid: 1 },
          { oid: 2 },
          { oid: 'TIMEOUT' },
          { oid: 'ERROR' },
        ];
        store.dispatch(sendAlarmAck(viewId, 'ACKID', alarms, 'COMMENT', ackType));
        jest.runAllTimers();
        expect(store.getActions()).toMatchSnapshot();
      });
    });
  });
});
