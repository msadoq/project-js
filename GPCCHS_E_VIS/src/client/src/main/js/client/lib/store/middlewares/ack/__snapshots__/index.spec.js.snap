// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ack:middleware GroundMonitoringAlarm mixed several success, failure and timeout 1`] = `
Array [
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "alarms": Array [
        Object {
          "oid": 1,
        },
        Object {
          "oid": 2,
        },
        Object {
          "oid": "TIMEOUT",
        },
        Object {
          "oid": "ERROR",
        },
      ],
      "comment": "COMMENT",
      "viewId": "c691a648-36c3-4c5a-b131-07aa86977437",
    },
    "type": "WS_VIEW_GMA_ALARM_ACK",
  },
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "oid": 1,
      "viewId": "c691a648-36c3-4c5a-b131-07aa86977437",
    },
    "type": "WS_VIEW_GMA_ALARM_ACK_SUCCESS",
  },
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "oid": 2,
      "viewId": "c691a648-36c3-4c5a-b131-07aa86977437",
    },
    "type": "WS_VIEW_GMA_ALARM_ACK_SUCCESS",
  },
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "error": "ERROR",
      "oid": "ERROR",
      "viewId": "c691a648-36c3-4c5a-b131-07aa86977437",
    },
    "type": "WS_VIEW_GMA_ALARM_ACK_FAILURE",
  },
  Object {
    "meta": Object {
      "jestSerialized": true,
    },
    "payload": Object {
      "containerId": "global",
      "messages": Array [
        Object {
          "content": "Acknowledging error : ERROR",
        },
      ],
      "type": "danger",
    },
    "type": "WS_MESSAGE_ADD",
  },
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "error": "Timeout",
      "oid": "TIMEOUT",
      "viewId": "c691a648-36c3-4c5a-b131-07aa86977437",
    },
    "type": "WS_VIEW_GMA_ALARM_ACK_FAILURE",
  },
  Object {
    "meta": Object {
      "jestSerialized": true,
    },
    "payload": Object {
      "containerId": "global",
      "messages": Array [
        Object {
          "content": "Acknowledging error : Timeout",
        },
      ],
      "type": "danger",
    },
    "type": "WS_MESSAGE_ADD",
  },
  Object {
    "meta": Object {
      "jestSerialized": true,
    },
    "payload": Object {
      "containerId": "global",
      "messages": Array [
        Object {
          "content": "2 alarms successfully acknowledged",
        },
      ],
      "type": "success",
    },
    "type": "WS_MESSAGE_ADD",
  },
]
`;

exports[`ack:middleware GroundMonitoringAlarm several ack failure 1`] = `
Array [
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "alarms": Array [
        Object {
          "oid": "ERROR",
        },
        Object {
          "oid": "ERROR",
        },
        Object {
          "oid": "ERROR",
        },
        Object {
          "oid": "ERROR",
        },
      ],
      "comment": "COMMENT",
      "viewId": "c691a648-36c3-4c5a-b131-07aa86977437",
    },
    "type": "WS_VIEW_GMA_ALARM_ACK",
  },
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "error": "ERROR",
      "oid": "ERROR",
      "viewId": "c691a648-36c3-4c5a-b131-07aa86977437",
    },
    "type": "WS_VIEW_GMA_ALARM_ACK_FAILURE",
  },
  Object {
    "meta": Object {
      "jestSerialized": true,
    },
    "payload": Object {
      "containerId": "global",
      "messages": Array [
        Object {
          "content": "Acknowledging error : ERROR",
        },
      ],
      "type": "danger",
    },
    "type": "WS_MESSAGE_ADD",
  },
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "error": "ERROR",
      "oid": "ERROR",
      "viewId": "c691a648-36c3-4c5a-b131-07aa86977437",
    },
    "type": "WS_VIEW_GMA_ALARM_ACK_FAILURE",
  },
  Object {
    "meta": Object {
      "jestSerialized": true,
    },
    "payload": Object {
      "containerId": "global",
      "messages": Array [
        Object {
          "content": "Acknowledging error : ERROR",
        },
      ],
      "type": "danger",
    },
    "type": "WS_MESSAGE_ADD",
  },
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "error": "ERROR",
      "oid": "ERROR",
      "viewId": "c691a648-36c3-4c5a-b131-07aa86977437",
    },
    "type": "WS_VIEW_GMA_ALARM_ACK_FAILURE",
  },
  Object {
    "meta": Object {
      "jestSerialized": true,
    },
    "payload": Object {
      "containerId": "global",
      "messages": Array [
        Object {
          "content": "Acknowledging error : ERROR",
        },
      ],
      "type": "danger",
    },
    "type": "WS_MESSAGE_ADD",
  },
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "error": "ERROR",
      "oid": "ERROR",
      "viewId": "c691a648-36c3-4c5a-b131-07aa86977437",
    },
    "type": "WS_VIEW_GMA_ALARM_ACK_FAILURE",
  },
  Object {
    "meta": Object {
      "jestSerialized": true,
    },
    "payload": Object {
      "containerId": "global",
      "messages": Array [
        Object {
          "content": "Acknowledging error : ERROR",
        },
      ],
      "type": "danger",
    },
    "type": "WS_MESSAGE_ADD",
  },
]
`;

exports[`ack:middleware GroundMonitoringAlarm several ack success 1`] = `
Array [
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "alarms": Array [
        Object {
          "oid": 1,
        },
        Object {
          "oid": 2,
        },
        Object {
          "oid": 3,
        },
        Object {
          "oid": 4,
        },
      ],
      "comment": "COMMENT",
      "viewId": "c691a648-36c3-4c5a-b131-07aa86977437",
    },
    "type": "WS_VIEW_GMA_ALARM_ACK",
  },
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "oid": 1,
      "viewId": "c691a648-36c3-4c5a-b131-07aa86977437",
    },
    "type": "WS_VIEW_GMA_ALARM_ACK_SUCCESS",
  },
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "oid": 2,
      "viewId": "c691a648-36c3-4c5a-b131-07aa86977437",
    },
    "type": "WS_VIEW_GMA_ALARM_ACK_SUCCESS",
  },
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "oid": 3,
      "viewId": "c691a648-36c3-4c5a-b131-07aa86977437",
    },
    "type": "WS_VIEW_GMA_ALARM_ACK_SUCCESS",
  },
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "oid": 4,
      "viewId": "c691a648-36c3-4c5a-b131-07aa86977437",
    },
    "type": "WS_VIEW_GMA_ALARM_ACK_SUCCESS",
  },
  Object {
    "meta": Object {
      "jestSerialized": true,
    },
    "payload": Object {
      "containerId": "global",
      "messages": Array [
        Object {
          "content": "4 alarms successfully acknowledged",
        },
      ],
      "type": "success",
    },
    "type": "WS_MESSAGE_ADD",
  },
]
`;

exports[`ack:middleware GroundMonitoringAlarm several ack timeout 1`] = `
Array [
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "alarms": Array [
        Object {
          "oid": "TIMEOUT",
        },
        Object {
          "oid": "TIMEOUT",
        },
        Object {
          "oid": "TIMEOUT",
        },
        Object {
          "oid": "TIMEOUT",
        },
      ],
      "comment": "COMMENT",
      "viewId": "c691a648-36c3-4c5a-b131-07aa86977437",
    },
    "type": "WS_VIEW_GMA_ALARM_ACK",
  },
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "error": "Timeout",
      "oid": "TIMEOUT",
      "viewId": "c691a648-36c3-4c5a-b131-07aa86977437",
    },
    "type": "WS_VIEW_GMA_ALARM_ACK_FAILURE",
  },
  Object {
    "meta": Object {
      "jestSerialized": true,
    },
    "payload": Object {
      "containerId": "global",
      "messages": Array [
        Object {
          "content": "Acknowledging error : Timeout",
        },
      ],
      "type": "danger",
    },
    "type": "WS_MESSAGE_ADD",
  },
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "error": "Timeout",
      "oid": "TIMEOUT",
      "viewId": "c691a648-36c3-4c5a-b131-07aa86977437",
    },
    "type": "WS_VIEW_GMA_ALARM_ACK_FAILURE",
  },
  Object {
    "meta": Object {
      "jestSerialized": true,
    },
    "payload": Object {
      "containerId": "global",
      "messages": Array [
        Object {
          "content": "Acknowledging error : Timeout",
        },
      ],
      "type": "danger",
    },
    "type": "WS_MESSAGE_ADD",
  },
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "error": "Timeout",
      "oid": "TIMEOUT",
      "viewId": "c691a648-36c3-4c5a-b131-07aa86977437",
    },
    "type": "WS_VIEW_GMA_ALARM_ACK_FAILURE",
  },
  Object {
    "meta": Object {
      "jestSerialized": true,
    },
    "payload": Object {
      "containerId": "global",
      "messages": Array [
        Object {
          "content": "Acknowledging error : Timeout",
        },
      ],
      "type": "danger",
    },
    "type": "WS_MESSAGE_ADD",
  },
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "error": "Timeout",
      "oid": "TIMEOUT",
      "viewId": "c691a648-36c3-4c5a-b131-07aa86977437",
    },
    "type": "WS_VIEW_GMA_ALARM_ACK_FAILURE",
  },
  Object {
    "meta": Object {
      "jestSerialized": true,
    },
    "payload": Object {
      "containerId": "global",
      "messages": Array [
        Object {
          "content": "Acknowledging error : Timeout",
        },
      ],
      "type": "danger",
    },
    "type": "WS_MESSAGE_ADD",
  },
]
`;

exports[`ack:middleware GroundMonitoringAlarm simple ack success 1`] = `
Array [
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "alarms": Array [
        Object {
          "oid": 1,
        },
      ],
      "comment": "COMMENT",
      "viewId": "c691a648-36c3-4c5a-b131-07aa86977437",
    },
    "type": "WS_VIEW_GMA_ALARM_ACK",
  },
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "oid": 1,
      "viewId": "c691a648-36c3-4c5a-b131-07aa86977437",
    },
    "type": "WS_VIEW_GMA_ALARM_ACK_SUCCESS",
  },
  Object {
    "meta": Object {
      "jestSerialized": true,
    },
    "payload": Object {
      "containerId": "global",
      "messages": Array [
        Object {
          "content": "1 alarm successfully acknowledged",
        },
      ],
      "type": "success",
    },
    "type": "WS_MESSAGE_ADD",
  },
]
`;

exports[`ack:middleware OnBoardAlarm mixed several success, failure and timeout 1`] = `
Array [
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "alarms": Array [
        Object {
          "oid": 1,
        },
        Object {
          "oid": 2,
        },
        Object {
          "oid": "TIMEOUT",
        },
        Object {
          "oid": "ERROR",
        },
      ],
      "comment": "COMMENT",
      "viewId": "225cce85-7d35-456c-97e9-be47a977e373",
    },
    "type": "WS_VIEW_OBA_ALARM_ACK",
  },
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "oid": 1,
      "viewId": "225cce85-7d35-456c-97e9-be47a977e373",
    },
    "type": "WS_VIEW_OBA_ALARM_ACK_SUCCESS",
  },
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "oid": 2,
      "viewId": "225cce85-7d35-456c-97e9-be47a977e373",
    },
    "type": "WS_VIEW_OBA_ALARM_ACK_SUCCESS",
  },
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "error": "ERROR",
      "oid": "ERROR",
      "viewId": "225cce85-7d35-456c-97e9-be47a977e373",
    },
    "type": "WS_VIEW_OBA_ALARM_ACK_FAILURE",
  },
  Object {
    "meta": Object {
      "jestSerialized": true,
    },
    "payload": Object {
      "containerId": "global",
      "messages": Array [
        Object {
          "content": "Acknowledging error : ERROR",
        },
      ],
      "type": "danger",
    },
    "type": "WS_MESSAGE_ADD",
  },
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "error": "Timeout",
      "oid": "TIMEOUT",
      "viewId": "225cce85-7d35-456c-97e9-be47a977e373",
    },
    "type": "WS_VIEW_OBA_ALARM_ACK_FAILURE",
  },
  Object {
    "meta": Object {
      "jestSerialized": true,
    },
    "payload": Object {
      "containerId": "global",
      "messages": Array [
        Object {
          "content": "Acknowledging error : Timeout",
        },
      ],
      "type": "danger",
    },
    "type": "WS_MESSAGE_ADD",
  },
  Object {
    "meta": Object {
      "jestSerialized": true,
    },
    "payload": Object {
      "containerId": "global",
      "messages": Array [
        Object {
          "content": "2 alarms successfully acknowledged",
        },
      ],
      "type": "success",
    },
    "type": "WS_MESSAGE_ADD",
  },
]
`;

exports[`ack:middleware OnBoardAlarm several ack failure 1`] = `
Array [
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "alarms": Array [
        Object {
          "oid": "ERROR",
        },
        Object {
          "oid": "ERROR",
        },
        Object {
          "oid": "ERROR",
        },
        Object {
          "oid": "ERROR",
        },
      ],
      "comment": "COMMENT",
      "viewId": "225cce85-7d35-456c-97e9-be47a977e373",
    },
    "type": "WS_VIEW_OBA_ALARM_ACK",
  },
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "error": "ERROR",
      "oid": "ERROR",
      "viewId": "225cce85-7d35-456c-97e9-be47a977e373",
    },
    "type": "WS_VIEW_OBA_ALARM_ACK_FAILURE",
  },
  Object {
    "meta": Object {
      "jestSerialized": true,
    },
    "payload": Object {
      "containerId": "global",
      "messages": Array [
        Object {
          "content": "Acknowledging error : ERROR",
        },
      ],
      "type": "danger",
    },
    "type": "WS_MESSAGE_ADD",
  },
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "error": "ERROR",
      "oid": "ERROR",
      "viewId": "225cce85-7d35-456c-97e9-be47a977e373",
    },
    "type": "WS_VIEW_OBA_ALARM_ACK_FAILURE",
  },
  Object {
    "meta": Object {
      "jestSerialized": true,
    },
    "payload": Object {
      "containerId": "global",
      "messages": Array [
        Object {
          "content": "Acknowledging error : ERROR",
        },
      ],
      "type": "danger",
    },
    "type": "WS_MESSAGE_ADD",
  },
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "error": "ERROR",
      "oid": "ERROR",
      "viewId": "225cce85-7d35-456c-97e9-be47a977e373",
    },
    "type": "WS_VIEW_OBA_ALARM_ACK_FAILURE",
  },
  Object {
    "meta": Object {
      "jestSerialized": true,
    },
    "payload": Object {
      "containerId": "global",
      "messages": Array [
        Object {
          "content": "Acknowledging error : ERROR",
        },
      ],
      "type": "danger",
    },
    "type": "WS_MESSAGE_ADD",
  },
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "error": "ERROR",
      "oid": "ERROR",
      "viewId": "225cce85-7d35-456c-97e9-be47a977e373",
    },
    "type": "WS_VIEW_OBA_ALARM_ACK_FAILURE",
  },
  Object {
    "meta": Object {
      "jestSerialized": true,
    },
    "payload": Object {
      "containerId": "global",
      "messages": Array [
        Object {
          "content": "Acknowledging error : ERROR",
        },
      ],
      "type": "danger",
    },
    "type": "WS_MESSAGE_ADD",
  },
]
`;

exports[`ack:middleware OnBoardAlarm several ack success 1`] = `
Array [
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "alarms": Array [
        Object {
          "oid": 1,
        },
        Object {
          "oid": 2,
        },
        Object {
          "oid": 3,
        },
        Object {
          "oid": 4,
        },
      ],
      "comment": "COMMENT",
      "viewId": "225cce85-7d35-456c-97e9-be47a977e373",
    },
    "type": "WS_VIEW_OBA_ALARM_ACK",
  },
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "oid": 1,
      "viewId": "225cce85-7d35-456c-97e9-be47a977e373",
    },
    "type": "WS_VIEW_OBA_ALARM_ACK_SUCCESS",
  },
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "oid": 2,
      "viewId": "225cce85-7d35-456c-97e9-be47a977e373",
    },
    "type": "WS_VIEW_OBA_ALARM_ACK_SUCCESS",
  },
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "oid": 3,
      "viewId": "225cce85-7d35-456c-97e9-be47a977e373",
    },
    "type": "WS_VIEW_OBA_ALARM_ACK_SUCCESS",
  },
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "oid": 4,
      "viewId": "225cce85-7d35-456c-97e9-be47a977e373",
    },
    "type": "WS_VIEW_OBA_ALARM_ACK_SUCCESS",
  },
  Object {
    "meta": Object {
      "jestSerialized": true,
    },
    "payload": Object {
      "containerId": "global",
      "messages": Array [
        Object {
          "content": "4 alarms successfully acknowledged",
        },
      ],
      "type": "success",
    },
    "type": "WS_MESSAGE_ADD",
  },
]
`;

exports[`ack:middleware OnBoardAlarm several ack timeout 1`] = `
Array [
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "alarms": Array [
        Object {
          "oid": "TIMEOUT",
        },
        Object {
          "oid": "TIMEOUT",
        },
        Object {
          "oid": "TIMEOUT",
        },
        Object {
          "oid": "TIMEOUT",
        },
      ],
      "comment": "COMMENT",
      "viewId": "225cce85-7d35-456c-97e9-be47a977e373",
    },
    "type": "WS_VIEW_OBA_ALARM_ACK",
  },
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "error": "Timeout",
      "oid": "TIMEOUT",
      "viewId": "225cce85-7d35-456c-97e9-be47a977e373",
    },
    "type": "WS_VIEW_OBA_ALARM_ACK_FAILURE",
  },
  Object {
    "meta": Object {
      "jestSerialized": true,
    },
    "payload": Object {
      "containerId": "global",
      "messages": Array [
        Object {
          "content": "Acknowledging error : Timeout",
        },
      ],
      "type": "danger",
    },
    "type": "WS_MESSAGE_ADD",
  },
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "error": "Timeout",
      "oid": "TIMEOUT",
      "viewId": "225cce85-7d35-456c-97e9-be47a977e373",
    },
    "type": "WS_VIEW_OBA_ALARM_ACK_FAILURE",
  },
  Object {
    "meta": Object {
      "jestSerialized": true,
    },
    "payload": Object {
      "containerId": "global",
      "messages": Array [
        Object {
          "content": "Acknowledging error : Timeout",
        },
      ],
      "type": "danger",
    },
    "type": "WS_MESSAGE_ADD",
  },
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "error": "Timeout",
      "oid": "TIMEOUT",
      "viewId": "225cce85-7d35-456c-97e9-be47a977e373",
    },
    "type": "WS_VIEW_OBA_ALARM_ACK_FAILURE",
  },
  Object {
    "meta": Object {
      "jestSerialized": true,
    },
    "payload": Object {
      "containerId": "global",
      "messages": Array [
        Object {
          "content": "Acknowledging error : Timeout",
        },
      ],
      "type": "danger",
    },
    "type": "WS_MESSAGE_ADD",
  },
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "error": "Timeout",
      "oid": "TIMEOUT",
      "viewId": "225cce85-7d35-456c-97e9-be47a977e373",
    },
    "type": "WS_VIEW_OBA_ALARM_ACK_FAILURE",
  },
  Object {
    "meta": Object {
      "jestSerialized": true,
    },
    "payload": Object {
      "containerId": "global",
      "messages": Array [
        Object {
          "content": "Acknowledging error : Timeout",
        },
      ],
      "type": "danger",
    },
    "type": "WS_MESSAGE_ADD",
  },
]
`;

exports[`ack:middleware OnBoardAlarm simple ack success 1`] = `
Array [
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "alarms": Array [
        Object {
          "oid": 1,
        },
      ],
      "comment": "COMMENT",
      "viewId": "225cce85-7d35-456c-97e9-be47a977e373",
    },
    "type": "WS_VIEW_OBA_ALARM_ACK",
  },
  Object {
    "payload": Object {
      "ackId": "ACKID",
      "oid": 1,
      "viewId": "225cce85-7d35-456c-97e9-be47a977e373",
    },
    "type": "WS_VIEW_OBA_ALARM_ACK_SUCCESS",
  },
  Object {
    "meta": Object {
      "jestSerialized": true,
    },
    "payload": Object {
      "containerId": "global",
      "messages": Array [
        Object {
          "content": "1 alarm successfully acknowledged",
        },
      ],
      "type": "success",
    },
    "type": "WS_MESSAGE_ADD",
  },
]
`;
